Source: https://github.com/rhinstaller/efivar/issues/60 , rebased against 34
Reason: Fix build on Exherbo
Upstream: No, seems specific to us

diff --git a/Make.defaults b/Make.defaults
index 0dd13c8..db209dd 100644
--- a/Make.defaults
+++ b/Make.defaults
@@ -18,9 +18,9 @@ CCLD_FOR_BUILD	?= $(CC_FOR_BUILD)
 CCLD	:= $(if $(filter undefined,$(origin CCLD)),$(CC),$(CCLD))
 OPTIMIZE ?= -O2 -flto
 CFLAGS	?= $(OPTIMIZE) -g3
-CFLAGS	:= $(CFLAGS)
+CFLAGS	:= $(CFLAGS) -D_GNU_SOURCE -D__USE_GNU
 LDFLAGS ?=
-LDFLAGS := $(LDFLAGS)
+LDFLAGS := $(LDFLAGS) -D_GNU_SOURCE -D__USE_GNU
 AR	:= $(CROSS_COMPILE)$(COMPILER)-ar
 NM	:= $(CROSS_COMPILE)$(COMPILER)-nm
 RANLIB	:= $(CROSS_COMPILE)$(COMPILER)-ranlib
@@ -29,8 +29,8 @@ ABIDIFF := abidiff
 
 PKGS	=
 
-clang_cflags = -D_GNU_SOURCE -std=gnu11 -Wno-address-of-packed-member
-gcc_cflags = -specs=$(TOPDIR)/gcc.specs
+clang_cflags = -D_GNU_SOURCE -D__USE_GNU -std=gnu11 -Wno-address-of-packed-member
+gcc_cflags = -specs=$(TOPDIR)/gcc.specs $(clang_cflags)
 cflags	= $(CFLAGS) -I${TOPDIR}/src/include/ \
 	$(if $(findstring clang,$(CC)),$(clang_cflags),) \
 	$(if $(findstring gcc,$(CC)),$(gcc_cflags),) \
diff --git a/gcc.specs b/gcc.specs
index 45d43d1..d91d97c 100644
--- a/gcc.specs
+++ b/gcc.specs
@@ -1,5 +1,5 @@
 *cpp:
-+ -D_GNU_SOURCE
++ -D_GNU_SOURCE -D__USE_GNU
 
 *efivar_cpp_options:
  -Werror -Wall -std=gnu11 -Wextra
